llama_stack/providers/utils/memory/vector_store.py

```diff
@@ -167,7 +167,9 @@ async def add_chunks(self, chunks: List[Chunk], embeddings: NDArray):
         raise NotImplementedError()
 
     @abstractmethod
-    async def query(self, embedding: NDArray, k: int, score_threshold: float) -> QueryChunksResponse:
+    async def query(
+        self, embedding: NDArray, k: int, score_threshold: float, query_str: Optional[str], search_mode: Optional[str]
```

Comment:
You use `Optional[str]` here but not in other places, we should probably be consistent. 

FWIW, I am preferential to adding type hints of this nature in general but that may be a broader point of discussion across the codebase.